code coverage report example

sebastian-lerner opened the pull request.

skaasten commented on jekyll/_cci2/rerun-failed-tests.adoc:
````diff
@@ -275,7 +275,46 @@ NOTE: The `halt` command will execute the rest of the _current_ step, regardless
+          when: always
+```
+
+The snippet above uses the built-in xref:caching[`.Revision`] key to keep all coverage reports from the same git commit (sha1) together.  On a rerun, both the original job run's coverage report and the rerun's coverage report will be uploaded as an artifact to `cover.html`. It can then be used in a downstream job for aggregation or analysis.
````
```suggestion
The snippet above uses the built-in xref:caching[`.Revision`] key to store a coverage report for the current vcs revision.  On a successful rerun, the original job run's coverage report will be restored to include the coverage from the skipped (passed) tests. It can then be used in a downstream job for aggregation or analysis.
```

skaasten commented on jekyll/_cci2/rerun-failed-tests.adoc:
```diff
@@ -275,7 +275,44 @@ NOTE: The `halt` command will execute the rest of the _current_ step, regardless
-* Jobs that upload code coverage reports link:https://discuss.circleci.com/t/product-launch-re-run-failed-tests-only-circleci-tests-run/47775/3?u=sebastian-lerner[may see issues during a rerun].
+* Jobs that upload code coverage reports:
+
+One way to ensure that code coverage reports from the original job run are persisted to an artifact in addition to the report that is generated on a re-run, see the following example for a sample Go project:
```
```suggestion
To ensure that code coverage reports from the original job run are persisted to an artifact in addition to the report that is generated on a re-run, see the following example for a sample Go project:
```

rosieyohannan commented on jekyll/_cci2/rerun-failed-tests.adoc:
```diff
@@ -270,12 +270,52 @@ steps:
 * Jobs that are older than the xref:persist-data#custom-storage-usage[retention period] for workspaces for the organization cannot be rerun with "Rerun failed tests".
-* Jobs that upload code coverage reports link:https://discuss.circleci.com/t/product-launch-re-run-failed-tests-only-circleci-tests-run/47775/3?u=sebastian-lerner[may see issues during a rerun].
+* Jobs that upload code coverage reports:
+
```
```suggestion
+
```

rosieyohannan commented on jekyll/_cci2/rerun-failed-tests.adoc:
```diff
@@ -270,12 +270,52 @@ steps:
+* Jobs that upload code coverage reports:
+
+To ensure that code coverage reports from the original job run are persisted to an artifact in addition to the report that is generated on a re-run, see the following example for a sample Go project:
+
```
```suggestion
+
```

rosieyohannan commented on jekyll/_cci2/rerun-failed-tests.adoc:
```diff
@@ -270,12 +270,52 @@ steps:
+----
+{% endraw %}
+
+The snippet above uses the built-in xref:caching[`.Revision`] key to store a coverage report for the current vcs revision.  On a successful rerun, the original job run's coverage report will be restored to include the coverage from the skipped (passed) tests. It can then be used in a downstream job for aggregation or analysis.
```
```suggestion
The snippet above uses the built-in xref:caching#[`.Revision`] key to store a coverage report for the current VCS revision.  On a successful rerun, the original job run's coverage report will be restored to include the coverage from the skipped (passed) tests. It can then be used in a downstream job for aggregation or analysis.
```

rosieyohannan commented on jekyll/_cci2/rerun-failed-tests.adoc:
```diff
@@ -270,12 +270,52 @@ steps:
+
+The snippet above uses the built-in xref:caching[`.Revision`] key to store a coverage report for the current vcs revision.  On a successful rerun, the original job run's coverage report will be restored to include the coverage from the skipped (passed) tests. It can then be used in a downstream job for aggregation or analysis.
+
+* This feature is not currently supported for the Windows execution environment.
```
```suggestion
* Rerun failed tests is not currently supported for the Windows execution environment.
```

sebastian-lerner commented:
added one more bit to remove a confusing thing that i added yesterday. pytest supports more than just name --timings-type=name, we don't need to specify it in here

rosieyohannan commented on jekyll/_cci2/rerun-failed-tests.adoc:
```diff
@@ -270,12 +270,52 @@ steps:
+          when: always
+----
+{% endraw %}
+
```
```suggestion
+
```
